lua/full_visual_line/internal.lua

```diff
@@ -12,7 +12,7 @@ end
 function M.setup_autocmd()
 	a.nvim_create_autocmd({ 'CursorMoved', 'ModeChanged' }, {
 		group = autocmd_group,
-		callback = M.handle_autocmd,
+		callback = function(opts) vim.schedule(function() M.handle_autocmd(opts) end) end,
```

Comment:
`vim.schedule_wrap` would be simpler

Reply:
Could you please elaborate on what syntax you're looking for here? 
`vim.schedule_wrap` takes only one argument, the callback, so you'd lose `opts` unless you also passed it like I'm doing here for `vim.schedule`

e.g.
Not allowed
```lua
callback = function(opts) vim.schedule_wrap(M.handle_autocmd, opts) end,
```

`opts` was lost
```lua
callback = function(opts) vim.schedule_wrap(M.handle_autocmd) end,
```

I don't use vim.schedule_wrap often so I don't know all of its details. My impression is that it's meant for scheduling inside of an already scheduled call. (https://youtu.be/GMS0JvS7W1Y?si=FR3JcufmDzbs4x7v&t=354)

Please let me know if I misunderstood and I'm happy to test if you have an exact call in mind for me to try.

Reply:
```lua
callback = vim.schedule_wrap(M.handle_autocmd),
```

Reply:
Oh okay cool, that works. Now I see what I was doing wrong. I forgot to remove the function(opts) so schedule_wrap was never called. Will change